cpmpy/solvers/exact.py

```diff
@@ -0,0 +1,492 @@
+#!/usr/bin/env python
+#-*- coding:utf-8 -*-
+##
+## exact.py
+##
+"""
+    Interface to Exact
+
+    Exact solves decision and optimization problems formulated as integer linear programs. Under the hood, it converts integer variables to binary (0-1) variables and applies highly efficient propagation routines and strong cutting-planes / pseudo-Boolean conflict analysis.
+
+    https://gitlab.com/JoD/exact
+
+    ===============
+    List of classes
+    ===============
+
+    .. autosummary::
+        :nosignatures:
+
+        CPM_exact
+"""
+import sys  # for stdout checking
+import time
+
+from .solver_interface import SolverInterface, SolverStatus, ExitStatus
+from ..expressions.core import *
+from ..expressions.variables import _BoolVarImpl, NegBoolView, _IntVarImpl, _NumVarImpl, intvar
+from ..transformations.comparison import only_numexpr_equality
+from ..transformations.flatten_model import flatten_constraint, flatten_objective, get_or_make_var
+from ..transformations.get_variables import get_variables
+from ..transformations.decompose_global import decompose_global
+from ..transformations.linearize import linearize_constraint, only_positive_bv
+from ..transformations.reification import only_bv_implies, reify_rewrite
+from ..expressions.globalconstraints import DirectConstraint
+import numpy as np
+
+class CPM_exact(SolverInterface):
+    """
+    Interface to the Python interface of Exact
+
+    Requires that the 'exact' python package is installed:
+    $ pip install exact
+    See https://pypi.org/project/exact for more information.
+
+    Creates the following attributes (see parent constructor for more):
+    xct_solver: the Exact instance used in solve()
+    """
+
+    @staticmethod
+    def supported():
+        # try to import the package
+        try:
+            import exact
+            return True
+        except ImportError as e:
+            return False
+
+
+    def __init__(self, cpm_model=None, subsolver=None):
+        """
+        Constructor of the native solver object
+
+        Requires a CPMpy model as input, and will create the corresponding
+        Exact solver object xct_solver.
+
+        xct_solver can be modified externally before
```

Comment:
we have a uniform way in the constructor to lists the properties tht are stored in this class, e.g. I see `xct_solver`, assum_dict, solv_inited, ...
please list exhaustively here

Reply:
maybe also why you store it, e.g. solve_minimize because it only supports minimisation so you have to manually flip the sign or so?

Reply:
looking at the file directly instead of the diff, I see that you do explain the xct_solver in the class docstring; so please add the others there too.

I now also see that you wrote about the inits that the objective can only be initialized once, which explains why you need these other attribs, so repeating that in the class docstring too
(this resolves multiple of my comments)

Reply:
Extended class docstring